presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -1100,6 +1102,26 @@ public boolean usesLegacyTableLayouts(Session session, TableHandle table)
                         Optional.empty()));
     }
 
+    @Override
+    public Optional<AggregatePushdownResult<TableHandle>> applyAggregation(Session session, TableHandle table, List<AggregateFunction> aggregates, Map<String, ColumnHandle> assignments, Optional<List<List<ColumnHandle>>> groupBy)
+    {
+        CatalogName catalogName = table.getCatalogName();
+        ConnectorMetadata metadata = getMetadata(session, catalogName);
+
+        if (metadata.usesLegacyTableLayouts()) {
+            return Optional.empty();
+        }
+
+        ConnectorSession connectorSession = session.toConnectorSession(catalogName);
+        return metadata.applyAggregation(connectorSession, table.getConnectorHandle(), aggregates, assignments, groupBy)
+                .map(result -> new AggregatePushdownResult<>(
```

Comment:
Here we need to verify the return data is sane & safe.
See https://github.com/prestosql/presto/pull/3324

Reply:
Added. Given the results are identical I extracted a common verification method.